app/com/m3/octoparts/cache/PartResponseCachingSupport.scala

```diff
@@ -76,9 +77,13 @@ trait PartResponseCachingSupport extends PartRequestServiceBase {
       }
       super.processWithConfig(ci, partRequestInfo, params)
     }
+    case e: HystrixRuntimeException => {
+      LTSVLogger.warn(e)
+      Future.failed(e) // Don't retry on HystrixRuntime exception
+    }
     case NonFatal(e) => {
-      LTSVLogger.error(e, "Memcached error" -> e.getClass.getSimpleName)
-      super.processWithConfig(ci, partRequestInfo, params)
+      LTSVLogger.error(e)
+      super.processWithConfig(ci, partRequestInfo, params) // Unknown error, retry
```

Comment:
I think it might be worthwhile to keep this around just to make sure that this layer of the onion is never allowed to fail. I'd rather have it try too hard and find out using Zipkin ..